### PR TITLE
Test and document all modern Python versions

### PR DESCRIPTION
**Pull request by jdufresne**
Add testing for Python 3.5 and 3.6.

Keep Travis and tox configuration in sync.

Document all versions using trove classifiers. Helps library users know if the package is compatible with a project.

### CODE REVIEW

**Comment by jdufresne** [comment]
Looks like Travis is not actually enabled for this project. You'll need to follow the getting started guide:

https://docs.travis-ci.com/user/getting-started/

**Comment by kmike** [comment]
That's strange, I'm pretty sure Travis is enabled. For example, see https://travis-ci.org/kmike/text-unidecode/builds/296979232.

**Comment by jdufresne** [comment]
Rebased to resolve merge conflicts.

Yup, you're right. Looks like Travis is running now. Thanks!

**Comment by jdufresne** [comment]
Python 3.2 is failing as it appears pytest does not support Python 3.2. Shall I simply drop Python 3.2 support? Python 3.2 is EOL since 2016-02-20 and so is not receiving bug or _security_ fixes. What do you think?

For details on supported Python versions, see:

https://docs.python.org/devguide/#status-of-python-branches

**Comment by kmike** [comment]
> Shall I simply drop Python 3.2 support?

Yes, +1 to drop Python 2.6 and 3.2 support.

**Comment by jdufresne** [comment]
Great! Done in the latest revision.

File: setup.py [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
 
 
-if sys.version_info >= (3, ):
-    with codecs.open('README.rst', encoding='utf-8') as f:
-        long_description = f.read()
-else:
-    with open('README.rst') as f:
-        long_description = f.read()
+with codecs.open('README.rst', encoding='utf-8') as f:
+    long_description = f.read()
```
Comment (kmike) [inline setup.py]: Hey, what was the reason for this change? I recall there were issues with pip/setuptools/distutils and non-ascii text in long_description, and using unicode in Python 3 and bytes in Python 2 was the closest thing to make it work reliably.

Comment (jdufresne) [reply setup.py]: I see, thanks. I mistakenly thought this was a workaround for Python 2.6, but it sounds like it is necessary for Python 2.7. I have reverted these lines with the latest revision.

**Comment by kmike** [comment]
Thanks @jdufresne for fixing the CI!

**Pull request merged**